kgraphql/src/main/kotlin/com/apurebase/kgraphql/schema/model/ast/DocumentNode.kt

```diff
@@ -1,7 +1,47 @@
 package com.apurebase.kgraphql.schema.model.ast
 
+import java.util.*
+
 data class DocumentNode(
     val loc: Location?,
     val definitions: List<DefinitionNode>
 
-)
+) {
+    val fields: List<String> by lazy {
+        val q: Queue<SelectionSetNode> = LinkedList<SelectionSetNode>()
+        val defs = this.definitions.mapNotNull { d: DefinitionNode ->
+            when (d) {
+                is DefinitionNode.ExecutableDefinitionNode ->
+                    d
+                else -> null
+            }
+        }
+        defs.forEach { d ->
+            q.add(d.selectionSet)
+        }
+
+        val fields = mutableListOf<NameNode>()
+        while (!q.isEmpty()) {
```

Comment:
Maybe having that BFS doesn't have sense, and we need to fetch only a single level?